src/cts/src/TreeBuilder.h

```diff
@@ -120,11 +143,34 @@ class TreeBuilder
                     double x2,
                     double y2)
   {
-    if ((x >= x1) && (x <= x2) && (y >= y1) && (y <= y2)) {
+    if ((x > x1) && (x < x2) && (y > y1) && (y < y2)) {
       return true;
     }
     return false;
   }
+  bool isAlongBbox(double x,
+                   double y,
+                   double x1,
+                   double y1,
+                   double x2,
+                   double y2)
+  {
+    if ((floatEqual(x, x1) || (floatEqual(x, x2)))
+        && (floatEqualOrGreater(y, y1) && floatEqualOrSmaller(y, y2))) {
```

Comment:
The main goal of integer DBU is to avoid writing code like this.  Can this be avoided?

Reply:
This may be hard because there are many arithmetic operations on these locations.  Let me see what can be done.